### PR TITLE
LPS-183167 Changing a profile client-extension yaml file should make certain tasks out of date

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-183167](https://issues.liferay.com/browse/LPS-183167).

### CODE REVIEW

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java [hunk 1]
```diff
@@ -677,7 +652,10 @@ private void _configureClientExtensionTasks(
 		TaskProvider<Zip> buildSiteInitializerZipTaskProvider,
 		TaskProvider<CreateClientExtensionConfigTask>
 			createClientExtensionConfigTaskProvider,
-		TaskProvider<DefaultTask> validateClientExtensionIdsTaskProvider) {
+		TaskProvider<DefaultTask> validateClientExtensionIdsTaskProvider,
```
Comment (petershin) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:655]: Use interface `Task`

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java [hunk 2]
```diff
@@ -663,9 +631,16 @@ private final Map<String, JsonNode> _configureClientExtensionJsonNodes(
...
 			profileJsonNodes.put(profileName, jsonNode);
 
-			_addInputFile(
-				file, () -> _isActiveProfile(project, profileName),
-				taskProviders);
+			for (TaskProvider<?> taskProvider : taskProviders) {
+				taskProvider.configure(
+					task -> {
+						if (_isActiveProfile(project, profileName)) {
+							TaskInputs inputs = task.getInputs();
```
Comment (petershin) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:638]: Use Java class name for clarity `taskInputs`

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java [hunk 3]
```diff
@@ -732,6 +720,15 @@ public String call() throws Exception {
 				validateClientExtensionIdsTask.setGroup(
 					LifecycleBasePlugin.VERIFICATION_GROUP);
 
+				TaskInputs inputs = validateClientExtensionIdsTask.getInputs();
+
+				inputs.file(clientExtensionYamlFile);
+
+				TaskOutputs outputs =
```
Comment (petershin) [inline modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:727]: Use Java class name for clarity `taskOutputs`

**Review by petershin — CHANGES_REQUESTED** [review]

File: modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java (hunk 1 above)
Comment (drewbrokke) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:655]: In order to change to `Task`, I'll have to change it to `Task` all the way up to the original declaration of the original `TaskProvider` like so:
From:
```java
		TaskProvider<DefaultTask> validateClientExtensionIdsTaskProvider =
			GradleUtil.addTaskProvider(
				project, VALIDATE_CLIENT_EXTENSION_IDS_TASK_NAME,
				DefaultTask.class);

		TaskProvider<DefaultTask> validateClientExtensionTaskProvider =
			GradleUtil.addTaskProvider(
				project, VALIDATE_CLIENT_EXTENSIONS_TASK_NAME,
				DefaultTask.class);
```

To:
```java
		TaskProvider<Task> validateClientExtensionIdsTaskProvider =
			GradleUtil.addTaskProvider(
				project, VALIDATE_CLIENT_EXTENSION_IDS_TASK_NAME,
				Task.class);

		TaskProvider<Task> validateClientExtensionTaskProvider =
			GradleUtil.addTaskProvider(
				project, VALIDATE_CLIENT_EXTENSIONS_TASK_NAME,
				Task.class);
```

Is that what you had in mind?

Comment (petershin) [reply modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java:655]: yup

**Comment by petershin** [comment]
ci:forward

**Comment by petershin** [comment]
ci:forward

**Review by petershin — APPROVED** [review]

**Pull request closed without merging**